src/sentry/release_health/metrics.py

```diff
@@ -458,165 +447,105 @@ def run_sessions_query(
-                    referrer="release_health.metrics.get_release_sessions_time_bounds.terminal_sessions",
-                    use_cache=False,
-                )["data"]
+
+        if environments:
+            where.append(
+                Condition(
+                    lhs=Column(name="tags[environment]"),
+                    op=Op.IN,
+                    rhs=environments,
+                )
             )
-        except MetricIndexNotFound:
-            pass
-
-        # This check is added because if there are no sessions found, then the
-        # aggregations query return both the sessions_lower_bound and the
-        # sessions_upper_bound as `0` timestamp and we do not want that behaviour
-        # by default
-        # P.S. To avoid confusion the `0` timestamp which is '1970-01-01 00:00:00'
-        # is rendered as '0000-00-00 00:00:00' in clickhouse shell
-        formatted_unix_start_time = datetime.utcfromtimestamp(0).strftime("%Y-%m-%dT%H:%M:%S+00:00")
 
-        lower_bound: Optional[str] = None
-        upper_bound: Optional[str] = None
+        query = MetricsQuery(
+            org_id=org_id,
+            project_ids=[project_id],
+            select=select,
+            start=datetime(2008, 5, 8, tzinfo=timezone.utc),  # Sentry first commit
+            end=datetime.now(timezone.utc) + timedelta(seconds=10),
+            where=where,
+            granularity=Granularity(LEGACY_SESSIONS_DEFAULT_ROLLUP),
+            include_series=False,
+            include_totals=True,
+        )
 
-        for row in rows:
-            if set(row.values()) == {formatted_unix_start_time}:
-                continue
-            if lower_bound is None or row["min"] < lower_bound:
-                lower_bound = row["min"]
-            if upper_bound is None or row["max"] > upper_bound:
-                upper_bound = row["max"]
+        raw_result = get_series(
+            projects=projects,
+            metrics_query=query,
+            use_case_id=USE_CASE_ID,
+        )
 
-        if lower_bound is None or upper_bound is None:
-            return {"sessions_lower_bound": None, "sessions_upper_bound": None}  # type: ignore
+        groups = raw_result["groups"]
 
         def iso_format_snuba_datetime(date: str) -> str:
             return datetime.strptime(date, "%Y-%m-%dT%H:%M:%S+00:00").isoformat()[:19] + "Z"
 
-        return {  # type: ignore
-            "sessions_lower_bound": iso_format_snuba_datetime(lower_bound),
-            "sessions_upper_bound": iso_format_snuba_datetime(upper_bound),
-        }
+        formatted_unix_start_time = datetime.utcfromtimestamp(0).strftime("%Y-%m-%dT%H:%M:%S+00:00")
+
+        def clean_date_string(d: Optional[str]) -> Optional[str]:
+            # This check is added because if there are no sessions found, then the
+            # aggregations query return both the sessions_lower_bound and the
+            # sessions_upper_bound as `0` timestamp and we do not want that behaviour
```

Comment:
```suggestion
            # sessions_upper_bound as `0` timestamp, and we do not want that behavior
```

Reply:
I prefer the british spelling and the oxford comma :-).

Reply:
sure.